velox/vector/tests/VectorTest.cpp

```diff
@@ -505,6 +505,16 @@ class VectorTest : public testing::Test, public test::VectorTestBase {
       EXPECT_TRUE(source->equalValueAt(target.get(), i, sourceSize + i));
     }
 
+    std::vector<BaseVector::CopyRange> ranges = {
```

Comment:
do we have a test that copies smaller portions of the source on different a target index somewhere in the middle of the target vector?

Reply:
This is already copying the source into the second half of the target.  In addition to that, for other complex types and strings, `copy` is using `copyRanges` under the hood so this is just to fill in the hole for row vector.